src/http/client.cc

```diff
@@ -340,6 +340,22 @@ future<> client::make_request(request req, reply_handler handle, reply::status_t
     return do_with(std::move(req), std::move(handle), [this, expected] (request& req, reply_handler& handle) mutable {
         return with_connection([this, &req, &handle, expected] (connection& con) {
             return do_make_request(con, req, handle, expected);
+        }).handle_exception_type([this, &req, &handle, expected] (const transport_error& ex) {
```

Comment:
I still think this auto-retry should be an option, not mandatory (and perhaps not even the default). I'm worried that not all Seastar users will think this behavior is appropriate.

Reply:
OK, I'll add bool-flag off by default. I don't have any strong arguments for having it always retry

Reply:
Added construction-time on/off switch (off by default)